### PR TITLE
[REVIEW] Feature/10 min to cudf

### PR DESCRIPTION
**Pull request by beckernick**
### Summary of Changes

This PR adds a "10 Minutes to cuDF" section to the documentation, analogous to the ["10 Minutes to Pandas"](https://pandas.pydata.org/pandas-docs/stable/10min.html) documentation. Examples are drawn from the cuDF cheatsheet, existing documentation, and the Pandas documentation.

To support this new documentation, the PR adds `ipython` to the docs requirements file (for the ipython sphinx extensions) and updates `conf.py` and `index.rst` accordingly.

This documentation is currently in its own `rst` file, though it could naturally be moved to the Developer Documentation section.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
cc @randerzander for documentation

File: docs/source/10min.rst [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+
+Object Creation
+---------------
+
+Creating a `Series`.
+
+.. ipython:: python
+
+    s = cudf.Series([1,2,3,None,4])
+    print(s)
```
Comment (harrism) [inline docs/source/10min.rst]: Can you show the output of each print / section?

Comment (beckernick) [reply docs/source/10min.rst]: The sphinx extension automatically generates the output when you make the rst file. We could include it as suppressed output/comments explicitly in the file if you think people will want to look at the raw .rst source file, but I think that would create clutter. Do you think auto-generating the output is enough?

Comment (beckernick) [reply docs/source/10min.rst]: For reference: https://github.com/pandas-dev/pandas/blob/master/doc/source/10min.rst

Comment (harrism) [reply docs/source/10min.rst]: Ah, didn't know that. This is fine. :)

**Comment by kkraus14** [comment]
> @randerzander does the [builddocs_py35.yml](https://github.com/rapidsai/cudf/blob/master/conda/environments/builddocs_py35.yml) file also need to include ipython in the pip section to auto-build the official docs?

We definitely need to install ipython but we can do so from conda instead of from pip. I will push the change to your PR now.

**Comment by beckernick** [comment]
Perfect. Thanks, Keith.

File: conda/environments/builddocs_py35.yml [hunk 2]
```diff
@@ -17,8 +17,8 @@ dependencies:
 - nvstrings
 - pandas=0.20.*
 - pyarrow=0.10.*
-- pip:
-  - numpydoc
-  - sphinx
-  - sphinx-rtd-theme
-  - sphinxcontrib-websupport
```
Comment (kkraus14) [inline conda/environments/builddocs_py35.yml]: @mike-wendt unsure if you'd know this but seeing as you've fought with RTD the most would moving these away from pip to conda be okay?

**Comment by kkraus14** [comment]
Merging, thanks @beckernick!

**Pull request merged**